Libraries/StyleSheet/ColorPropType.js

```diff
@@ -9,26 +9,31 @@
   * @providesModule ColorPropType
   */
 'use strict';
+
 var ReactPropTypes = require('ReactPropTypes');
+
 var tinycolor = require('tinycolor');
 
-var colorValidator = function (props, propName) {
+var ColorPropType = function(props, propName) {
   var selectedColor = props[propName];
-  if (selectedColor === null || selectedColor === undefined || selectedColor.toString().trim() === '') {
-    return new Error(
-      `Invalid argument supplied to ${propName}.Expected a string like #123ADF or 'red'.`
-    );
+  if (selectedColor === undefined) {
+    return;
   }
 
-  if (tinycolor(selectedColor.toString().trim()).isValid()) {
-    return null;
+  if (tinycolor(selectedColor) === null) {
+    return new Error(
+`Invalid color supplied to ${propName}: ${selectedColor}. Valid color formats are
+  - '#f0f' (#rgb)
+  - '#ff00ff' (#rrggbb)
+  - '#ff00ff00' (#aarrggbb)
```

Comment:
I think it was already #rrggbbaa - at least, that's how I remember implementing it (which was different from the original tinycolor implementation), so it won't be a breaking change if we keep that order.

And yes, it's frustrating that it's different from the order of the int32 we send to native, but unfortunately Android expects the byte order to be argb, and doesn't do any native-side post-processing (iOS doesn't care, as we convert it to four floats on the native side anyway).